### PR TITLE
VAULT-32330: fix sanitize path function to account for backslashes

### PR DESCRIPTION
**Pull request by aslamovamir**
### Description
What does this PR do?
Fixes a backslash check in logical_system.go by expanding the check to also include filtering for double forward and back slashes.

### TODO only if you're a HashiCorp employee
- [x] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [x] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [x] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [x] **RFC:** If this change has an associated RFC, please link it in the description.
- [x] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: vault/logical_system.go
Comment (VioletHynes) [inline vault/logical_system.go:6172]: This might not be the correct based on my reading of the finding, i.e.
`has a leading slash, but not that it does not have '/' or '\' in its second position.`

For example, I think this would modify `\string` to `string`, but that it shouldn't. Based on my reading, I think this should only trim `/`, `//`, or `/\`.

**Review by VioletHynes — COMMENTED** [review]
A little comment so far. Remember also to add a title to the PR to briefly describe what it does, and a changelog will also need to be added to this PR.

I think the milestone is incorrect, as 1.16.0 has already released. My recommendation for a PR like this would be to use the `pr/no-milestone` tag for this PR as it shouldn't block any release. I would recommend only using milestones when this PR must be in an upcoming release.

**Comment by aslamovamir** [comment]
> A little comment so far. Remember also to add a title to the PR to briefly describe what it does, and a changelog will also need to be added to this PR.
> 
> I think the milestone is incorrect, as 1.16.0 has already released. My recommendation for a PR like this would be to use the `pr/no-milestone` tag for this PR as it shouldn't block any release. I would recommend only using milestones when this PR must be in an upcoming release.

Ahh nice notes, thanks Violet! 
I will add the changelog, does it make sense to label it as `security` type in the changelog?
Ahh I see your point about the milestone. The ticket mentions the fix to be backported to all relevant versions (1.18, 1.17, 1.16). How do we add correct labels to backport again?

**Comment by VioletHynes** [comment]
I would label this as an `improvement` or `bug` -- you can look around for examples to base yours off!

For backporting, just search `backport/` in labels and add all three of them. Though you don't need to do that yet, you can wait until the PR is merged and it will work then too.

File: vault/logical_system.go
Comment (divyaac) [reply vault/logical_system.go:6172]: I might be misunderstanding - the way I understood this was that we not only have to check for a leading "/" but also a "//" and "/\". Is this what you're saying as well Violet?

Comment (VioletHynes) [reply vault/logical_system.go:6172]: My reading is that it needs to check for `/`, `//`, and `/\` -- it shouldn't modify a path that starts with `\`

Comment (divyaac) [reply vault/logical_system.go:6172]: Violet's comment is the way I understand it too

Comment (divyaac) [reply vault/logical_system.go:6172]: For clarity here, the behavior we want is:

//my_string ==> my_string
/my_string ==> my_string
/\my_string ==> my_string
\/my_string ==> \/my_string
\\\\\//my_string ==> \\\\\//my_string

Comment (aslamovamir) [reply vault/logical_system.go]: Pushed a commit to address the concerns, please do let me know if it looks better now, many thanks!

File: vault/logical_system.go
Comment (VioletHynes) [inline vault/logical_system.go:6172]: This seems to satisfy the requirements, it may be worth double checking with Mickael or somebody to make sure this is accurate, since the code scanner seems to still be complaining. I'll hook you two up :)

**Review by VioletHynes — APPROVED** [review]
Approved on my end, but we should double check before merging in case the scanner warning is legit.

**Review by mickael-hc — APPROVED** [review]
After reviewing the changes, the finding is a false positive and we can dismiss it

**Pull request merged**
